Java/corecrypto/src/main/java/com/breadwallet/corecrypto/Transfer.java

```diff
@@ -72,6 +78,15 @@ private Transfer(BRCryptoTransfer core, Wallet wallet) {
         this.amountSupplier = Suppliers.memoize(() -> Amount.create(core.getAmount()));
         this.directedSupplier = Suppliers.memoize(() -> Amount.create(core.getAmountDirected()));
         this.directionSupplier = Suppliers.memoize(() -> Utilities.transferDirectionFromCrypto(core.getDirection()));
+
+        attributesSupplier = Suppliers.memoize(() -> {
+            Set<TransferAttribute> attributes = new HashSet<>();
+            UnsignedLong count = core.getAttributeCount();
+            for (UnsignedLong i = UnsignedLong.ZERO; i.compareTo(count) < 0; i = i.plus(UnsignedLong.ONE)) {
+                attributes.add(com.breadwallet.corecrypto.TransferAttribute.create (core.getAttributeAt(i).get()));
```

Comment:
To manage memory here, you'd have to do something like:
`attributes.add(com.breadwallet.corecrypto.TransferAttribute.create (core.getAttributeAt(i).get()).copy());`

And then implement `copy()` in `TransferAttribute` such that it called the core copy and its own `TransferAttribute::create` such that the reference is tracked and cleaned.

Alternatively, just have `getAttributeAt()` (or write `copyAttributeAt()`) do the memory management for you.

Reply:
I did something similar in the wallet.getTransferAttributesFor() method.

Reply:
Like this:
```
       attributesSupplier = Suppliers.memoize(() -> {
            Set<TransferAttribute> attributes = new HashSet<>();
            UnsignedLong count = core.getAttributeCount();
            for (UnsignedLong i = UnsignedLong.ZERO; i.compareTo(count) < 0; i = i.plus(UnsignedLong.ONE)) {
                Optional<TransferAttribute> attribute = core.getAttributeAt(i)
                        .transform(TransferAttribute::create); // Uses the 'take' from '...AttributeAt'
                if (attribute.isPresent())
                    attributes.add(attribute.get().copy());
            }
            return attributes;
        });
```